subsys/net/lib/download_client/src/download_client.c

```diff
@@ -376,8 +376,14 @@ void download_thread(void *client, void *a, void *b)
 		LOG_DBG("Receiving up to %d bytes at %p...",
 			(sizeof(dl->buf) - dl->offset), (dl->buf + dl->offset));
 
-		len = recv(dl->fd, dl->buf + dl->offset,
-			   sizeof(dl->buf) - dl->offset, 0);
+		size_t recv_len = sizeof(dl->buf) - dl->offset;
+
+		if (dl->file_size) {
```

Comment:
I tend to agree with @lemrey here, I'm not convinced it's necessary. The server shall only send as much data as it originally announced, if it sends more then it puts the validity of the whole image received under a question.

Reply:
Sure. I do agree with both of your comments.

Looking at this from an attackers perspective. I have some concerns about this being abused by an attacker but at least the image write will fail if it goes out of bounds.

I did see at one point if you are able to make it so that device is able to download > filesize you can get some weird undefined behavior.